src/app/views/query-runner/query-input/auto-complete/AutoComplete.tsx

```diff
@@ -72,373 +75,182 @@ class AutoComplete extends Component<IAutoCompleteProps, IAutoCompleteState> {
+    if (!requestUrl) {
+      dispatch(fetchAutoCompleteOptions('', queryVersion));
       return;
     }
-    const param = url.split('$').pop()!.split('=')[0];
-    const section = parametersWithVerb.values.find((k: { name: string; }) => {
-      return k.name === `$${param}`;
-    });
 
-    if (section && section.items && section.items.length > 0) {
-      this.setSuggestions(section.items);
+    const urlExistsInStore = autoCompleteOptions && requestUrl === autoCompleteOptions.url;
+    if (urlExistsInStore) {
+      displayAutoCompleteSuggestions(autoCompleteOptions.url);
+      return;
     }
+    dispatch(fetchAutoCompleteOptions(requestUrl, queryVersion, context));
   }
 
-  private setSuggestions(suggestions: string[], compare?: string) {
-    const sortedSuggestions = this.sortSuggestions(suggestions);
-    this.setState({
-      filteredSuggestions: sortedSuggestions,
-      suggestions: sortedSuggestions,
-      showSuggestions: (suggestions.length > 0),
-      compare: compare || ''
-    });
-  }
-
-  private sortSuggestions(suggestions: string[]): string[] {
-    return suggestions.sort(dynamicSort(null, SortOrder.ASC));
-  }
-
-  public componentDidUpdate = (prevProps: IAutoCompleteProps) => {
-    if (prevProps.autoCompleteOptions !== this.props.autoCompleteOptions) {
-      if (this.props.autoCompleteOptions) {
-        this.performLocalSearch(this.state.userInput);
-      }
-    }
-
-    const newUrl = this.props.sampleQuery.sampleUrl;
-    if ((this.state.queryUrl === prevProps.sampleQuery.sampleUrl) && newUrl !== this.state.queryUrl) {
-      if (newUrl !== this.state.queryUrl) {
-        this.setState({
-          queryUrl: newUrl,
-          userInput: newUrl,
-          multiline: this.isOverflowing(newUrl)
-        });
-      }
-    }
-  }
+  const displayAutoCompleteSuggestions = (url: string) => {
 
+    const { index } = getLastDelimiterInUrl(url);
+    const { previous: preceedingText } = getSearchText(url, index!);
+    const shouldSuggestVersions = preceedingText === GRAPH_URL + '/';
 
-  private requestForAutocompleteOptions(url: string) {
-    const signature = sanitizeQueryUrl(url);
-    const { requestUrl, queryVersion } = parseSampleUrl(signature);
-    if (GRAPH_API_VERSIONS.includes(queryVersion.toLowerCase())) {
-      if (!requestUrl) {
-        this.props.actions!.fetchAutoCompleteOptions('', queryVersion);
-      } else {
-        if (!this.props.autoCompleteOptions || `${requestUrl}` !== this.props.autoCompleteOptions.url) {
-          this.props.actions!.fetchAutoCompleteOptions(requestUrl, queryVersion);
-        } else {
-          this.performLocalSearch(url);
-        }
+    const theSuggestions = (shouldSuggestVersions) ? GRAPH_API_VERSIONS : getSuggestions(url, autoCompleteOptions!);
+    if (theSuggestions.length > 0) {
+      const filtered = (searchText) ? getFilteredSuggestions(searchText, theSuggestions) : theSuggestions;
+      if (filtered[0] !== searchText) {
```

Comment:
What's the effect of showing suggestions when the searchText is the first index?
I don't understand what this is achieving

Reply:
This fixes an edge case where the query parameters ($select .. etc) even after being selected would still show up as suggestions instead of showing the next round of suggestions after the parameters